File: php/24-swap-nodes-in-pairs.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution 
{
    /**
     * @param ListNode $head
     * @return ListNode
     */
    public function swapPairs($head) 
	{
        $dummy = new ListNode(0);
        $dummy->next = $head;
        $node = $dummy;
        while ($node->next != null) {
            if (isset($node->next->next)) {
                $next3 = $node->next->next->next;
                $next2 = $node->next->next;
                $next1 = $node->next;
                $node->next = $next2;
                $next2->next = $next1;
                $next1->next = $next3;
            }
            $node = $node->next->next;
        }
        return $dummy->next;
    }
}

File: go/24-swap-nodes-in-pairs.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	dummy := &ListNode{Next: head}
	node := dummy
	for node.Next != nil {
		if node.Next.Next != nil {
			n1 := node.Next
			n2 := node.Next.Next
			n3 := node.Next.Next.Next
			node.Next = n2
			n2.Next = n1
			n1.Next = n3
			node = node.Next.Next
		} else {
			break
		}

	}
	return dummy.Next
}
